### PR TITLE
Cake 2302 - rewrite nav response on fc-enabled community api

### PR DESCRIPTION
**Pull request by nmonterroso**
@Wikia/cake 
https://wikia-inc.atlassian.net/browse/CAKE-2302

Rewrites the output from `NavApiController.getData()` with the data from the CGS. I'm also enforcing the limits MW has on nav levels since the sitemap from CGS is the full sitemap. I created https://wikia-inc.atlassian.net/browse/CAKE-2308 to change this on the CGS side (while still allowing sitemap editing to happen on the FC).

### CODE REVIEW

File: extensions/wikia/FandomCreator/FandomCreatorApi.php [hunk 1]
```diff
@@ -14,10 +16,32 @@ class FandomCreatorApi {
...
-	public function __construct($baseUrl, $communityId) {
+	public function __construct(string $baseUrl, string $communityId) {
 		$this->baseUrl = $baseUrl;
 		$this->communityId = $communityId;
 	}
-}
\ No newline at end of file
+
+	public function getCommunity() {
+		$response = $this->doApiRequest("{$this->baseUrl}/communities/{$this->communityId}");
```
Comment (WikiaTech) [inline extensions/wikia/FandomCreator/FandomCreatorApi.php]: MAJOR Remove this unused "$response" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/FandomCreator/FandomCreatorApi.php [hunk 2]
```diff
@@ -14,10 +16,32 @@ class FandomCreatorApi {
...
-	public function __construct($baseUrl, $communityId) {
+	public function __construct( string $baseUrl, string $communityId ) {
 		$this->baseUrl = $baseUrl;
 		$this->communityId = $communityId;
 	}
-}
\ No newline at end of file
+
+	public function getCommunity() {
+		$response = $this->doApiRequest( "{$this->baseUrl}/communities/{$this->communityId}" );
```
Comment (WikiaTech) [inline extensions/wikia/FandomCreator/FandomCreatorApi.php]: MAJOR Remove this unused "$response" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

File: extensions/wikia/FandomCreator/Hooks.php [hunk 3]
```diff
@@ -9,11 +9,59 @@
...
+
+	private static function convertToSitemapData( $entry, $currentLevel, $maxElementsPerLevel ) {
+		$numLevels = count( $maxElementsPerLevel );
+		if ( $currentLevel > $numLevels ) {
+			return null;
+		}
+
+		$data = [
+				'text' => $entry->name,
+				'href' => self::getEntityPath( $entry->id )
```
Comment (lizlux) [inline extensions/wikia/FandomCreator/Hooks.php:46]: indentation looks off here

Comment (nmonterroso) [reply extensions/wikia/FandomCreator/Hooks.php:46]: I ran this through the MW stylize script, so I think it's correct.

Comment (lizlux) [reply extensions/wikia/FandomCreator/Hooks.php:46]: oh weird okay

**Pull request merged**